x/interchainstaking/keeper/abci.go

```diff
@@ -16,7 +16,26 @@ type zoneItrFn func(index int64, zoneInfo types.Zone) (stop bool)
 // BeginBlocker of interchainstaking module
 func (k Keeper) BeginBlocker(ctx sdk.Context) {
 	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyBeginBlocker)
-
+	// post upgrade-v1.2.5 processing
+	if ctx.BlockHeight() == 14540740 {
+		k.IterateReceipts(ctx, func(_ int64, receiptInfo types.Receipt) (stop bool) {
+			if receiptInfo.ChainId == "regen-1" && receiptInfo.Completed == nil {
+				sendMsg := banktypes.MsgSend{
+					FromAddress: "",
+					ToAddress:   "",
+					Amount:      receiptInfo.Amount,
+				}
+				zone, found := k.GetZone(ctx, "regen-1")
+				if found {
+					err := k.handleSendToDelegate(ctx, &zone, &sendMsg, receiptInfo.Txhash)
+					if err != nil {
+						panic(err)
+					}
+				}
+			}
+			return false
+		})
```

Comment:
Let's factor this into a separate function (perhaps `k.v010205postUpgradeHandler()`).

This will let us add a test for the function.

Reply:
`handleSendTodelegate` does a ica-delegate-tx - testing which requires ica channels on the fly - in current testing suite which doesnt seem to be possible - e2e-test would help in this

Reply:
keeper_test.go does have some functions for configuring ICA channels (the local end of at least) to allow k.SubmitTx to pass.

Reply:
If its possible to mock / set them up on the fly, then we should do that